src/App.tsx

```diff
@@ -102,12 +106,22 @@ const Content = () => {
 
   const { dialogAreaRef } = useDialogArea() ?? {};
 
+  const { data: referralAddress, isSuccess: isReferralAddressSuccess } = useReferralAddress(
+    testFlags.referralCode
```

Comment:
Since we are relying on the query param, I don't think `useReferralAddress` requires a param. We can access `testFlags` or query param directly in the hook. I also don't think that `testFlags` is the most appropriate place for `referralCode`

Reply:
i think its fine to pass in the referralcode as a param to useReferralAddress, since that is just a simple wrapper around the HTTP fetch call and should try to be as unopinionated about react state as possible

Reply:
but i moved all of this logic out of App.tsx into its own hook to keep this file as simple as possible